native_store.go

```diff
@@ -46,23 +48,25 @@ type nativeStore struct {
 //   - https://docs.docker.com/engine/reference/commandline/login#credentials-store
 func NewNativeStore(helperSuffix string) Store {
 	return &nativeStore{
-		programFunc: client.NewShellProgramFunc(remoteCredentialsPrefix + helperSuffix),
+		exe: executer.NewExecuter(remoteCredentialsPrefix + helperSuffix),
 	}
 }
 
 // Get retrieves credentials from the store for the given server.
-func (ns *nativeStore) Get(_ context.Context, serverAddress string) (auth.Credential, error) {
+func (ns *nativeStore) Get(ctx context.Context, serverAddress string) (auth.Credential, error) {
 	var cred auth.Credential
-	dockerCred, err := client.Get(ns.programFunc, serverAddress)
+	out, err := ns.exe.Execute(ctx, strings.NewReader(serverAddress), "get")
 	if err != nil {
-		if credentials.IsErrCredentialsNotFound(err) {
-			// do not return an error if the credentials are not in the keychain.
-			return auth.EmptyCredential, nil
-		}
+		return auth.EmptyCredential, err // some handling needed
+	}
+	dockerCred := &credentials.Credentials{
```

Comment:
Should I implement this struct and completely remove the `docker-credentials-helper` dependency?

Reply:
We should completely remove the `docker-credentials-helper` dependency

Reply:
Implemented the struct and removed the dependency.